pygen/pygen_src/test/riscv_instr_cov_test.py

```diff
@@ -0,0 +1,136 @@
+                                pass
+                            if not self.sample():
+                                if not expect_illegal_instr:
+                                    logging.error("Found unexpected illegal instr: {} [{}]".format(self.trace["instr"], entry))
+                                    self.unexpected_illegal_instr_cnt += 1
+                            # TODO: call sample here to do the functional coverage, and move assign_trace_info into it
+                            #self.sample(instruction)
+                        self.entry_cnt += 1
+                    line_count += 1
+                logging.info("[{}]: {} instruction processed".format(csv_file, self.entry_cnt))
+                self.total_entry_cnt += self.entry_cnt
+        logging.info("Finished processing {} trace CSV, {} instructions".format(len(self.trace_csv), self.total_entry_cnt))
+        if self.skipped_cnt > 0 or self.unexpected_illegal_instr_cnt > 0:
+            logging.error("{} instruction skipped, {} illegal instructions".format(self.skipped_cnt), self.unexpected_illegal_instr_cnt)
+
+    def post_process_trace(self):
+        pass
+
+    def sample(self):
+        instr_name, binary = "", ""
+        get_val(self.trace["binary"], binary, hexa=1)
+        if binary[-2:] != "11": #TODO: and RV32C in supported_isa 
+            #TODO: sample compressed instruction
+            pass
+        if binary[-2:] == "11":
+            #TODO: sampling
+            pass
+        #TODO: buch of if statements to check if the instruction name is valid and is a member of registered ones
+        instr_name = self.trace["instr"]
+        instruction = riscv_instr()
+        #TODO: check the instruction group...
+        self.assign_trace_info_to_instr(instruction)
+        #TODO: instruction.pre_sample() and sample(instruction)
+        return True
+
+    def assign_trace_info_to_instr(self, instruction):
+        operands, gpr_update, pair = {}, {}, {}
+        instruction.pc = get_val(self.trace["pc"], instruction.pc, hexa=1)
+        instruction.binary = get_val(self.trace["binary"], instruction.binary, hexa=1)
+        instruction.instr = self.trace["instr"]
+        instruction.gpr = self.trace["gpr"]
+        instruction.csr = self.trace["csr"]
+        instruction.mode = self.trace["mode"]
+        instruction.trace = self.trace["instr_str"]
+        instruction.operands = self.trace["operand"]
+        instruction.pad = self.trace["pad"]
+        #TODO: instr.update_src_regs(instr.operands)
+        #TODO: instr.update_dst_regs()
+
+def parse_args():
+    parse = argparse.ArgumentParser()
+    parse.add_argument("--csv_dir", help="directory consists of the CSV files", type=str, default = ".")
+    args = parse.parse_args()
+    return args
+
+def main():
+    parse = argparse.ArgumentParser()
+    parse.add_argument("--csv_dir", help="directory consists of the CSV files", type=str, default = ".")
+    args = parse.parse_args()
```

Comment:
do you mean to call the `parse_args()` defined above instead?

Reply:
sorry, I take back this comment actually.

it's actually unnecessary for this coverage test to search for all the `.csv` files that are generated, as the `cov.py` script already gets the pathname to the CSV filenames [here](https://github.com/google/riscv-dv/blob/master/cov.py#L137-L147) and passes them all into the coverage test as cmdline arguments - you should be able to just iterate through `sys.argv` and pull the pathnames from there directly.